src/components/shared/editable-text-input-list/editable-text-input-list.tsx

```diff
@@ -3,37 +3,65 @@
  * See the LICENSE file in the repository root folder for details.
  */
 
-import { Button, Form, InputGroup } from "react-bootstrap";
+import { Button, Form, InputGroup } from 'react-bootstrap';
+import PageBlock from '../page-block/page-block';
 
 interface EditableTextInputListProps {
+  title: string;
   items: any[];
+  readOnly?: boolean;
+  ordered?: boolean;
   add: (val: any) => any;
   remove: (idx: number) => any;
   update: (idx: number, val: any) => any;
 }
 
 const EditableTextInputList: React.FC<EditableTextInputListProps> = ({
+  title,
   items,
+  readOnly,
+  ordered,
   add,
   remove,
   update
 }) => {
+  let listPrepared = items.map((item, index: number) =>
+    !readOnly ? (
+      <div key={index} className={'mb-2'}>
+        <InputGroup>
+          <Form.Control
+            type="text"
+            value={item.toString()}
+            onChange={(e) => {
+              update(index, e.target.value);
+            }}
+          />
+          <Button type="button" variant="danger" onClick={() => remove(index)}>
+            X
+          </Button>
+        </InputGroup>
+      </div>
+    ) : (
+      <li key={index}>{item}</li>
+    )
+  );
+
+  const addButton = (
+    <Button type="button" variant="success" onClick={() => add('')}>
+      + Add New Bullet
+    </Button>
+  );
+
+  if (!readOnly) {
+    listPrepared = [...listPrepared, addButton];
+  }
+  let builtList = <ul>{listPrepared}</ul>;
+  if (ordered) {
+    builtList = <ol>{listPrepared}</ol>;
+  }
+
   return (
-    <>
-      {
-        items.map((item, index: number) => (
-          <div key={index} className={'mb-2'}>
-            <InputGroup>
-              <Form.Control type='text' value={item.toString()} onChange={(e) => {
-                update(index, e.target.value);
-              }} />
-              <Button type='button' variant='danger' onClick={() => remove(index)}>X</Button>
-            </InputGroup>
-          </div>
-        ))
-      }
-      <Button type='button' variant='success' onClick={() => add('')}>+ Add New Bullet</Button>
-    </>
+    <PageBlock title={title} headerRight={<></>} body={<Form.Group>{builtList}</Form.Group>} />
```

Comment:
I don't think it's a good idea to wrap the content of the `EditableTextInputList` in a `PageBlock`, reason being that it would make the styling very weird if we wanted to have multiple `EditableTextInputList`s in a single `PageBlock` or some other component. It does mean that there is more work to do in other higher-level components, but that also means this component can be easily swapped out in the future if we decide to make a new implementation of `EditableTextInputList` (which I hope we do because I made this very quickly so there are quite a few issues with it) or decide we don't want to use it anymore

Reply:
Thoughts on adding a PageBlock prop? So it can be rendered in it's own pageblock or without it

Reply:
If you add a PageBlock prop, that would be tightly coupling this component to the PageBlock component, which is what I'm trying to avoid. Even if we end up continuing to use PageBlock's for our UI, I think it's still bad design to have components reliant on each other in this manner, especially with how versatile the use cases for an EditableTextInputList can be.